src/Evolution/DiscontinuousGalerkin/MortarData.cpp

```diff
@@ -56,6 +58,134 @@ void MortarData<Dim>::insert_neighbor_mortar_data(
 }
 
+template <size_t Dim>
+void MortarData<Dim>::insert_local_geometric_quantities(
+    const Scalar<DataVector>& local_volume_det_inv_jacobian,
+    const Scalar<DataVector>& local_face_det_jacobian,
+    const Scalar<DataVector>& local_face_normal_magnitude) noexcept {
+  ASSERT(local_mortar_data_.has_value(),
+         "Must set local mortar data before setting the geometric quantities.");
+  ASSERT(local_face_det_jacobian[0].size() ==
+             local_face_normal_magnitude[0].size(),
+         "The determinant of the local face Jacobian has "
+             << local_face_det_jacobian[0].size()
+             << " grid points, and the magnitude of the local face normal has "
+             << local_face_normal_magnitude[0].size()
+             << " but they must be the same");
+  ASSERT(local_face_det_jacobian[0].size() ==
+             std::get<0>(*local_mortar_data()).number_of_grid_points(),
+         "The number of grid points ("
+             << std::get<0>(*local_mortar_data()).number_of_grid_points()
+             << ") on the local face must match the number of grid points "
+                "passd in for the face Jacobian determinant and normal vector "
+                "magnitude ("
+             << local_face_det_jacobian[0].size() << ")");
+  const size_t required_storage_size = local_volume_det_inv_jacobian[0].size() +
+                                       2 * local_face_det_jacobian[0].size();
+  if (local_volume_and_face_det_jacobian_and_normal_magnitude_.size() !=
+      required_storage_size) {
+    local_volume_and_face_det_jacobian_and_normal_magnitude_.resize(
+        required_storage_size);
+  }
+  std::copy(local_volume_det_inv_jacobian[0].begin(),
+            local_volume_det_inv_jacobian[0].end(),
+            local_volume_and_face_det_jacobian_and_normal_magnitude_.begin());
+  std::copy(
+      local_face_det_jacobian[0].begin(), local_face_det_jacobian[0].end(),
+      local_volume_and_face_det_jacobian_and_normal_magnitude_.begin() +
+          static_cast<std::ptrdiff_t>(local_volume_det_inv_jacobian[0].size()));
+  std::copy(
+      local_face_normal_magnitude[0].begin(),
+      local_face_normal_magnitude[0].end(),
+      local_volume_and_face_det_jacobian_and_normal_magnitude_.begin() +
+          static_cast<std::ptrdiff_t>(local_volume_det_inv_jacobian[0].size() +
+                                      local_face_det_jacobian[0].size()));
+}
+
+template <size_t Dim>
+void MortarData<Dim>::insert_local_face_normal_magnitude(
+    const Scalar<DataVector>& local_face_normal_magnitude) noexcept {
+  ASSERT(local_mortar_data_.has_value(),
+         "Must set local mortar data before setting the local face normal.");
+  const size_t required_storage_size = local_face_normal_magnitude[0].size();
+  if (local_volume_and_face_det_jacobian_and_normal_magnitude_.size() !=
+      required_storage_size) {
+    local_volume_and_face_det_jacobian_and_normal_magnitude_.resize(
+        required_storage_size);
+  }
```

Comment:
A few complicating factors (also more thinking out loud):
- the `MortarData` is mutated and so needs to be resized. 
- I _think_ subcell makes this quite challenging because in that case you need some additional quantities on the boundary, but I don't know yet how many. Currently, I think everything is needed for the subcell that touches the neighbor, since it's effectively doing one Gauss point DG (i.e. FV). What this means is that when doing subcell, independent of the quadrature some history of the additional quantities is probably needed.

So what I think wouldn't work (and that you didn't suggest) is using the quadrature stored in the Mesh. But I think the bool would work since for any given simulation you need to be using one or the other, you cannot mix them (or, mixing would be a risky optimization). I think adding the bool flags is a good way to go